build_runner/lib/src/generate/build_impl.dart

```diff
@@ -189,9 +189,13 @@ class BuildImpl {
   }
 
   Set<AssetId> _inputsForPhase(int phaseNumber) => _assetGraph.allNodes
-      .where((n) =>
-          n is! GeneratedAssetNode ||
-          (n as GeneratedAssetNode).phaseNumber < phaseNumber)
+      .where((n) {
+        if (n is GeneratedAssetNode) {
+          return n.wasOutput == true && n.phaseNumber < phaseNumber;
```

Comment:
are we worried about null values in `wasOutput`? We don't have this in other places... would prefer this without the `== true`